### PR TITLE
feat!: add disease-normalizer data sources

### PR DESCRIPTION
**Pull request by jsstevenson**
I've built out working classes for both disease and therapy normalization, but since disease is giving us trouble I wanted to put this PR up first. It makes some changes:

* Add OncoTree, DO, NCIt source classes
* Add a "custom" data source class. A user provides version and download methods, and they get the data storage, file selection, etc routines for free. I've used this to a) define some custom OMIM rules for how we do things in the disease normalizer (we could also write an internal tool to grab stuff from IGM annotations, I just wasn't sure about writing that out in an open repo) and b) manage the extra RxNorm file that we generate in therapy.
* move helper functions into the `utils` module
* other miscellaneous simplification

WIP disease implementation is here: https://github.com/cancervariants/disease-normalization/pull/163

Here's an example of the `CustomData` class for handling RxNorm drug forms: https://github.com/cancervariants/therapy-normalization/blob/d5794d6e9c83d3a4ad1dc2f56be166d5c63dcaf5/therapy/etl/rxnorm.py#L71

### CODE REVIEW

File: src/wags_tails/chembl.py [hunk 1]
```diff
@@ -19,14 +17,13 @@ class ChemblData(DataSource):
...
-            $WAGS_TAILS_DIR, or within a "wags_tails" subdirectory under environment
-            variables $XDG_DATA_HOME or $XDG_DATA_DIRS, or finally, at
-            ``~/.local/share/``
+        :param data_dir: direct location to store data files in, if specified. See
+            ``get_data_dir()`` in the ``storage_utils`` module for further configuration
+            details.
         :param silent: if True, don't print any info/updates to console
         """
         self._src_name = "chembl"
+        self._filetype = "db"
```
Comment (korikuzma) [inline src/wags_tails/chembl.py:26]: I think these should be class variables (similar to`DataSource`)

File: src/wags_tails/do.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    def __init__(self, data_dir: Optional[Path] = None, silent: bool = False) -> None:
+        """Set common class parameters.
+
+        :param data_dir: direct location to store data files in, if specified. See
+            ``get_data_dir()`` in the ``storage_utils`` module for further configuration
+            details.
+        :param silent: if True, don't print any info/updates to console
+        """
+        self._src_name = "do"
+        self._filetype = "owl"
```
Comment (korikuzma) [inline src/wags_tails/do.py:27]: I think these should be class variables (similar to`DataSource`)

File: src/wags_tails/ncit.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    def __init__(self, data_dir: Optional[Path] = None, silent: bool = False) -> None:
+        """Set common class parameters.
+
+        :param data_dir: direct location to store data files in, if specified. See
+            ``get_data_dir()`` in the ``storage_utils`` module for further configuration
+            details.
+        :param silent: if True, don't print any info/updates to console
+        """
+        self._src_name = "ncit"
+        self._filetype = "owl"
```
Comment (korikuzma) [inline src/wags_tails/ncit.py:24]: I think these should be class variables (similar to`DataSource`)

File: src/wags_tails/oncotree.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+        :param data_dir: direct location to store data files in. If not provided, tries
+            to find a "oncotree" subdirectory within the path at environment variable
+            $WAGS_TAILS_DIR, or within a "wags_tails" subdirectory under environment
+            variables $XDG_DATA_HOME or $XDG_DATA_DIRS, or finally, at
+            ``~/.local/share/``
+        :param silent: if True, don't print any info/updates to console
+        """
+        self._src_name = "oncotree"
+        self._filetype = "json"
```
Comment (korikuzma) [inline src/wags_tails/oncotree.py:30]: I think these should be class variables (similar to`DataSource`)

File: src/wags_tails/utils/downloads.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+_logger = logging.getLogger(__name__)
+
+
+def handle_zip(dl_path: Path, outfile_path: Path) -> None:
+    """Provide simple callback function to extract the largest file within a given
+    zipfile and save it within the appropriate data directory.
+
+    :param Path dl_path: path to temp data file
+    :param Path outfile_path: path to save file within
```
Comment (korikuzma) [inline src/wags_tails/utils/downloads.py:21]:
```suggestion
    :param dl_path: path to temp data file
    :param outfile_path: path to save file within
```

File: src/wags_tails/utils/downloads.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    tqdm_params: Optional[Dict] = None,
+) -> None:
+    """Perform HTTP download of remote data file.
+
+    :param url: URL to retrieve file from
+    :param outfile_path: path to where file should be saved. Must be an actual
+        Path instance rather than merely a pathlike string.
+    :param headers: Any needed HTTP headers to include in request
+    :param handler: provide if downloaded file requires additional action, e.g.
+        it's a zip file.
```
Comment (korikuzma) [inline src/wags_tails/utils/downloads.py:48]: Missing tqdm_params in docstring

File: src/wags_tails/utils/downloads.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        r.raise_for_status()
+        total_size = int(r.headers.get("content-length", 0))
+        with open(dl_path, "wb") as h:
+            if not tqdm_params["disable"]:
+                if "apiKey" in url:
+                    pattern = r"&apiKey=.{8}-.{4}-.{4}-.{4}-.{12}"
+                    print_url = re.sub(pattern, "", os.path.basename(url))
+                    print(f"Downloading {print_url}")
+                else:
+                    print(f"Downloading {os.path.basename(url)}")
```
Comment (korikuzma) [inline src/wags_tails/utils/downloads.py:68]: I think we typically use ellipsis for in-progress tasks

```suggestion
                    print(f"Downloading {print_url}...")
                else:
                    print(f"Downloading {os.path.basename(url)}...")
```

File: src/wags_tails/chembl.py (hunk 1 above)
Comment (jsstevenson) [reply src/wags_tails/chembl.py]: 👍 good call

**Review by korikuzma — APPROVED** [review]
🚀

**Pull request merged**